File: app/src/main/java/com/yannis/thesis/movierecommendationapp/api/APIService.java
package com.yannis.thesis.movierecommendationapp.api;

import com.yannis.thesis.movierecommendationapp.models.DirectorResponse;
import com.yannis.thesis.movierecommendationapp.models.GenreResponse;
import com.yannis.thesis.movierecommendationapp.models.Movie;
import com.yannis.thesis.movierecommendationapp.models.MovieResponse;
import com.yannis.thesis.movierecommendationapp.models.PrimaryMovieInfo;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.Path;
import retrofit2.http.Query;

/**
 * Created by yiannos on 20-Feb-18.
 */

public interface APIService {

    @POST("/list")
    Call<Movie> loadMovie();

    @GET("movie/top_rated")
    Call<MovieResponse> getTopRatedMovies(@Query("api_key") String apiKey);

    @GET("movie/popular")
    Call<MovieResponse> getPopularMovies(@Query("api_key") String apiKey);

    @GET("movie/{id}")
    Call<Movie> getMovieDetails(@Path("id")Integer id, @Query("api_key") String apiKey);

    @GET("search/movie")
    Call<MovieResponse> getMovieByTitle(@Query("query") String title, @Query("api_key") String apiKey);

    @GET("discover/movie")
    Call<MovieResponse> getMovieByReleasedYear(@Query("primary_release_year") String year, @Query("api_key") String apiKey);

    @GET("search/person")
    Call<DirectorResponse> getPersonIdByName(@Query("query") String name, @Query("api_key") String apiKey);

    @GET("discover/movie")
    Call<MovieResponse> getMovieByDirector(@Query("with_crew") int directorId, @Query("api_key") String apiKey);

    //https://api.themoviedb.org/3/genre/movie/list?api_key=efbdebf1b30ffab728c49495748e9dfa
    @GET("genre/movie/list")
    Call<GenreResponse> getAllGenres(@Query("api_key")String apiKey);

    @GET("discover/movie")
    Call<MovieResponse> getMovieByGenre(@Query("with_genres") int genre, @Query("api_key") String apiKey);

    //@Query("with_cast")
    @GET("discover/movie")
    Call<MovieResponse> getMovieByActor(@Query("with_cast") int castId, @Query("api_key") String apiKey);




}

File: app/src/main/java/com/yannis/thesis/movierecommendationapp/api/APIService.kt
package com.yannis.thesis.movierecommendationapp.api

import com.yannis.thesis.movierecommendationapp.models.DirectorResponse
import com.yannis.thesis.movierecommendationapp.models.GenreResponse
import com.yannis.thesis.movierecommendationapp.models.Movie
import com.yannis.thesis.movierecommendationapp.models.MovieResponse
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.Path
import retrofit2.http.Query

interface APIService {
    @POST("/list")
    fun loadMovie(): Call<Movie>

    @GET("movie/top_rated")
    fun getTopRatedMovies(@Query("api_key") apiKey: String): Call<MovieResponse>

    @GET("movie/popular")
    fun getPopularMovies(@Query("api_key") apiKey: String): Call<MovieResponse>

    @GET("movie/{id}")
    fun getMovieDetails(@Path("id") id: Int, @Query("api_key") apiKey: String): Call<Movie?>

    @GET("search/movie")
    fun getMovieByTitle(@Query("query") title: String, @Query("api_key") apiKey: String): Call<MovieResponse>

    @GET("discover/movie")
    fun getMovieByReleasedYear(
        @Query("primary_release_year") year: String,
        @Query("api_key") apiKey: String
    ): Call<MovieResponse>

    @GET("search/person")
    fun getPersonIdByName(@Query("query") name: String, @Query("api_key") apiKey: String): Call<DirectorResponse>

    @GET("discover/movie")
    fun getMovieByDirector(@Query("with_crew") directorId: Int, @Query("api_key") apiKey: String): Call<MovieResponse>

    @GET("genre/movie/list")
    fun getAllGenres(@Query("api_key") apiKey: String): Call<GenreResponse>

    @GET("discover/movie")
    fun getMovieByGenre(@Query("with_genres") genre: Int, @Query("api_key") apiKey: String): Call<MovieResponse>

    @GET("discover/movie")
    fun getMovieByActor(@Query("with_cast") castId: Int, @Query("api_key") apiKey: String): Call<MovieResponse>
}
